packages/synapse-react-client/src/components/ChallengeTeamWizard/ChallengeTeamWizard.tsx

```diff
@@ -79,366 +59,368 @@ export type ChallengeTeamWizardProps = {
+          buttonTooltip =
+            'You have already submitted a request to join this team.'
+        } else if (
+          selectedTeamMembershipStatus &&
+          selectedTeamMembershipStatus.membershipApprovalRequired
+        ) {
+          // The user has to send a request to join the selected team
+          buttonText = 'Request to Join Team'
+          buttonOnClickBehavior = () => {
+            setStep(ChallengeTeamWizardStep.JOIN_REQUEST_FORM)
+          }
+        } else if (
+          selectedTeamMembershipStatus &&
+          selectedTeamMembershipStatus.canJoin
+        ) {
+          // The user can freely join the selected team
+          buttonText = 'Join Team'
+          buttonOnClickBehavior = () => {
+            void addUserToTeam()
+          }
+        }
         return {
-          canJoin: false,
-          errorMessage: 'You are already a member of this team.',
+          content: challenge && !isMemberOfSubmissionTeam && (
+            <SelectChallengeTeam
+              challengeId={challenge.id}
+              onCreateTeam={() =>
+                setStep(ChallengeTeamWizardStep.CREATE_NEW_TEAM)
+              }
+              selectedTeamId={selectedTeamId}
+              onSelectTeam={teamId => setSelectedTeamId(teamId)}
+            />
+          ),
+          actions: (
+            <>
+              {backButton}
+              <Tooltip title={buttonTooltip}>
+                {/* Wrap button in span to show tooltip when button is disabled */}
+                <span>
+                  <Button
+                    onClick={buttonOnClickBehavior}
+                    startIcon={
+                      addMemberToTeamIsPending ? <SynapseSpinner /> : undefined
+                    }
+                    disabled={
+                      !selectedTeamId ||
+                      selectedTeamMembershipStatusIsLoading ||
+                      addMemberToTeamIsPending ||
+                      disableJoiningSelectedTeam
+                    }
+                    variant={'contained'}
+                  >
+                    {buttonText}
+                  </Button>
+                </span>
+              </Tooltip>
+            </>
+          ),
         }
       }
-
-      if (hasOpenRequest) {
-        // User already has an open request to join this team, disable Next button to avoid request spamming
+      case ChallengeTeamWizardStep.ACCEPT_INVITATION:
         return {
-          canJoin: false,
-          errorMessage:
-            'Your previous request to join this team is pending review.',
+          content: <OpenMembershipInvitation teamId={selectedTeamId!} />,
+          actions: (
+            <>
+              {backButton}
+              <AcceptMembershipInvitationButton
+                teamId={selectedTeamId!}
+                onSuccess={() => {
+                  setStep(ChallengeTeamWizardStep.REGISTRATION_SUCCESSFUL)
+                }}
+              />
+            </>
+          ),
```

Comment:
This is a new step to show a membership invitation and prompt the user to accept it